src/main/java/com/google/devtools/build/lib/skyframe/IgnoredPackagePrefixesFunction.java

```diff
@@ -105,30 +115,46 @@ public SkyValue compute(SkyKey key, Environment env)
             return null;
           }
           if (patternFileValue.isFile()) {
-            getIgnoredPackagePrefixes(rootedPatternFile, ignoredPackagePrefixesBuilder);
+            getIgnoredPackagePrefixes(
+                    rootedPatternFile,
+                    ignoredPackagePrefixesBuilder,
+                    env.getListener());
+            workspaceRoot = rootedPatternFile.getRoot();
           }
         }
       }
     }
 
-    return IgnoredPackagePrefixesValue.of(ignoredPackagePrefixesBuilder.build());
+    return IgnoredPackagePrefixesValue.of(ignoredPackagePrefixesBuilder.build(), workspaceRoot);
   }
 
   private static final class PathFragmentLineProcessor
       implements LineProcessor<ImmutableSet<PathFragment>> {
-    private final ImmutableSet.Builder<PathFragment> fragments = ImmutableSet.builder();
+    private final ImmutableSet.Builder<PathFragment> patterns = ImmutableSet.builder();
+    private final EventHandler eventHandler;
+
+    public PathFragmentLineProcessor(EventHandler eventHandler) {
+      this.eventHandler = eventHandler;
+    }
 
     @Override
     public boolean processLine(String line) {
-      if (!line.isEmpty() && !line.startsWith("#")) {
-        fragments.add(PathFragment.create(line));
+      String error = null;
+      if (line.isEmpty() || line.startsWith("#")) {
+        return true;
+      }
+      error = UnixGlob.checkPatternForError(line);
+      if (error != null) {
+        eventHandler.handle(Event.warn((error + " (in .bazelignore glob pattern '" + line + "')")));
```

Comment:
is it really ok to have errors here that don't fail the build?

Reply:
I can change it to throw exception or `INFO` if thats better. Let me know what you think, thanks!